tests/integration/test_charm.py

```diff
@@ -127,6 +127,18 @@ def peers(self):
     assert action.results["return"] == "null"
 
 
+@pytest.mark.abort_on_fail
+async def test_loki_endpoint(loki_relation_data):
+    """
+    arrange: after Maubot is deployed and relations established
+    act: any-loki is deployed and joins the relation
+    assert: logging settings in the relation data bag has a logging endpoint.
+    """
+    # Assert that the Loki endpoint is in the relation data
+    assert "endpoint" in loki_relation_data, "relation data are missing 'endpoint'"
+    assert "url" in loki_relation_data["endpoint"], "Relation data missing 'url'"
```

Comment:
```suggestion
@pytest.mark.abort_on_fail
async def test_loki_endpoint(ops_test, any_loki):
    """
    arrange: after Maubot is deployed and relations established
    act: any-loki is deployed and joins the relation
    assert: logging settings in the relation data bag has a logging endpoint.
    """
    assert "loki" in await ops_test.juju(
        "juju", "ssh", "--container", "maubot", "maubot/0", "pebble", "plan"
    )
```

The test here actually checks the integration data content set by the `any-charm`, which is unrelated to the maubot charm. I would recommend testing the corresponding response in the maubot charm instead.

Reply:
addressed